ReactiveDataDisplayManagerTests/REFACTORING/Stack/Manager/BaseStackManagerTests.swift

```diff
@@ -0,0 +1,188 @@
+        let generator6 = MockStackCellGenerator(title: "After Two")
+        let generator7 = MockStackCellGenerator(title: "After Two")
+        let generator8 = MockStackCellGenerator(title: "After Two")
+        let generatorsAfterTwo = [generator5, generator6, generator7, generator8]
+
+        // when
+        ddm.addCellGenerators(generators)
+        ddm.addCellGenerators(generatorsAfterTwo, after: generator2)
+        ddm.forceRefill()
+
+        // then
+        XCTAssertEqual(stackView.arrangedSubviews[2], generator5.view)
+        XCTAssertEqual(stackView.arrangedSubviews[3], generator6.view)
+        XCTAssertEqual(stackView.arrangedSubviews[4], generator7.view)
+        XCTAssertEqual(stackView.arrangedSubviews[5], generator8.view)
+    }
+
+    func testUpdatingGenerators() {
+        // given
+        let generator1 = MockStackCellGenerator(title: "One")
+        let generator2 = MockStackCellGenerator(title: "Two")
+        let generator3 = MockStackCellGenerator(title: "Three")
+        let generator4 = MockStackCellGenerator(title: "Four")
+        let generators = [generator1, generator2, generator3, generator4]
+
+        // when
+        ddm.addCellGenerators(generators)
+        ddm.forceRefill()
+
+        generator2.title = "Updated Two"
+        generator4.title = "Updated Four"
+        ddm.update(generators: [generator2, generator4])
+        ddm.forceRefill()
+
+        // then
+        XCTAssertEqual(stackView.arrangedSubviews.count, generators.count)
+        XCTAssertEqual((stackView.arrangedSubviews[1] as! UILabel).text, "Updated Two")
+        XCTAssertEqual((stackView.arrangedSubviews[3] as! UILabel).text, "Updated Four")
+    }
+
+    func testForceRefill() {
+        // given
+        let generator1 = MockStackCellGenerator(title: "One")
+        let generator2 = MockStackCellGenerator(title: "Two")
+        let generator3 = MockStackCellGenerator(title: "Three")
+        let generator4 = MockStackCellGenerator(title: "Four")
+        let generators = [generator1, generator2, generator3, generator4]
+
+        // when
+        ddm.addCellGenerators(generators)
+        ddm.forceRefill()
+
+        // then
+        XCTAssertEqual(stackView.arrangedSubviews.count, generators.count)
+    }
+
+    func testClearing() {
+        // given
+        let generator1 = MockStackCellGenerator(title: "One")
+        let generator2 = MockStackCellGenerator(title: "Two")
+        let generator3 = MockStackCellGenerator(title: "Three")
+        let generator4 = MockStackCellGenerator(title: "Four")
+        let generators = [generator1, generator2, generator3, generator4]
+
+        // when
+        ddm.addCellGenerators(generators)
+        ddm.forceRefill()
+
+        ddm.clearCellGenerators()
+        ddm.forceRefill()
+
+        // then
+        XCTAssertEqual(stackView.arrangedSubviews.count, 0)
+    }
+
+}
+
+// MARK: - Mocks
+
+fileprivate final class MockStackCellGenerator: StackCellGenerator {
```

Comment:
```suggestion
fileprivate final class StubStackCellGenerator: StackCellGenerator {
```

И в отдельный файл их вынести бы.